packages/core/src/snapshot.js

```diff
@@ -323,6 +324,70 @@ function waitForDiscoveryNetworkIdle(page, options) {
 // Used to cache resources across core instances
 const RESOURCE_CACHE_KEY = Symbol('resource-cache');
 
+function* triggerResourceRequests(page, snapshot) {
+  // copy widths to prevent mutation later
+  let widths = snapshot.widths.slice();
+  let initialWidth = widths.shift();
+
+  // set the initial page size
+  yield page.resize({
+    width: initialWidth,
+    height: snapshot.minHeight
+  });
+
+  // navigate to the url
+  yield page.goto(snapshot.url);
+
+  if (snapshot.execute) {
+    // when any execute options are provided, inject snapshot options
+    /* istanbul ignore next: cannot detect coverage of injected code */
+    yield page.eval((_, s) => (window.__PERCY__.snapshot = s), snapshot);
+    yield page.evaluate(snapshot.execute.afterNavigation);
+  }
+
+  // resize and set pixel ratio for higher density images
+  if (snapshot.deviceScaleFactor !== 1) {
+    yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
+    yield page.evaluate(snapshot.execute?.beforeResize);
+
+    yield page.resize({
+      mobile: true,
+      width: initialWidth,
+      height: snapshot.minHeight,
+      deviceScaleFactor: snapshot.deviceScaleFactor
+    });
+
+    yield page.reload();
+    yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
+    yield page.evaluate(snapshot.execute?.afterResize);
+  }
+
+  // trigger resize events for other widths
+  for (let width of widths) {
+    yield page.evaluate(snapshot.execute?.beforeResize);
+    yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
+    yield page.resize({ width, height: snapshot.minHeight });
+    yield page.evaluate(snapshot.execute?.afterResize);
+
+    // resize and set pixel ratio for higher density images
+    if (snapshot.deviceScaleFactor !== 1) {
+      yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
+      yield page.evaluate(snapshot.execute?.beforeResize);
+
+      yield page.resize({
+        width,
+        mobile: true,
+        height: snapshot.minHeight,
+        deviceScaleFactor: snapshot.deviceScaleFactor
+      });
+
+      yield page.reload();
+    } else {
+      yield waitForDiscoveryNetworkIdle(page, snapshot.discovery);
```

Comment:
Is this actually necessary? We already wait for idle at the top of the next loop, or after this whole function is called. So this line will essentially double idle time after resizing without mobile.

Reply:
Turns out it's not. I was a bit overzealous with this waits because at one point I was chasing flakes 😅